activiti-api-process-model/src/main/java/org/activiti/api/process/model/BPMNSignal.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.api.process.model;
+
+import org.activiti.api.process.model.payloads.SignalPayload;
+
+public interface BPMNSignal extends BPMNElement {
+
+	String getSignalName();
+	SignalPayload getSignalPayload();
```

Comment:
It sounds a little bit strange to me having both `getSignalName` and `getSignalPayload` at this level, because SignalPayload has already the signal name. IMO you should either flatten the content of the payload here (`name` and `variables`) or have only `getSignalPayload`. What do you think?

Reply:
Actually at the beginning I was thinking about flatten the content of payload, but what if it will be extended? getSignalName is kind of a helper function (internally it s using signalPayload).
I have no problems to delete it from interface definition, but will leave it in the implementation?

Reply:
Yes, it should be ok to keep it on the implementation.